### PR TITLE
Modernize team edit description flow

### PR DESCRIPTION
**Pull request by jakob223**

### CODE REVIEW

**Comment by jakob223** [comment]
Only the last commit is relevant.

**Comment by jakob223** [comment]
Bump @keybase/react-hackers . I promise there's only like one or two hundred lines of actual diff, the rest is merge artifacts.

File: shared/constants/types/teams.tsx [hunk 1]
```diff
@@ -114,38 +114,42 @@ export type TeamRoleMap = {
   loadedVersion: number
   roles: Map<TeamID, TeamRoleAndDetails>
 }
-
-export type State = Readonly<{
-  addUserToTeamsState: AddUserToTeamsState
-  addUserToTeamsResults: string
-  canPerform: Map<TeamID, TeamOperations>
+type Errors = {
```
Comment (chrisnojima) [inline shared/constants/types/teams.tsx]: i think pulling this out doesn't really help discoverability. it'll make TS popovers more obscure etc. if you want to 'collect' them maybe preview with error* instead

File: shared/teams/edit-team-description/index.tsx [hunk 2]
```diff
@@ -1,57 +1,100 @@
...
-  teamname: string
-  waitingKey: string
-}
+type Props = Container.RouteProps<{teamID: Types.TeamID}>
 
 const EditTeamDescription = (props: Props) => {
-  const {origDescription, teamname, onClose, onSubmit, waitingKey} = props
+  const teamID = Container.getRouteProps(props, 'teamID', Types.noTeamID)
+
+  const teamname = Container.useSelector(state => Constants.getTeamNameFromID(state, teamID))
```
Comment (chrisnojima) [inline shared/teams/edit-team-description/index.tsx]: likely easier to just pull state out vs calling useselector multiple times

Comment (jakob223) [reply shared/teams/edit-team-description/index.tsx]: As discussed in person, going to leave this as is.

**Pull request merged**